### PR TITLE
new custom hooks and callback files extracted from Taxi.js

### PR DESCRIPTION
**Pull request by sevak-yer**

### CODE REVIEW

File: src/callbacks/findNDI.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function findNDI(response) {
```
Comment (vnersisyan-vineti) [inline src/callbacks/findNDI.js:1]: what is NDI? Nearest driver? also please change to anonymous function

File: src/callbacks/findNDI.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  const distanceMatrixArray = response.rows[0].elements;
+  const distanceValues = distanceMatrixArray.map((value) => value.distance.value);
```
Comment (vnersisyan-vineti) [inline src/callbacks/findNDI.js:3]: separate these lines, put a blank line

File: src/callbacks/findNDI.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  distanceValues.sort((a, b) => a - b);
```
Comment (vnersisyan-vineti) [inline src/callbacks/findNDI.js:4]: separate these lines

File: src/callbacks/findNDI.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  const nearestdriverIndex = distanceMatrixArray.findIndex(
+    (elem) => elem.distance.value === distanceValues[0],
+  );
```
Comment (vnersisyan-vineti) [inline src/callbacks/findNDI.js:7]: separate these lines

File: src/callbacks/findReservationPrice.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+function findReservationPrice(response, state) {
+  const distanceMatrixArrayClient = response.rows[0].elements;
+  const distanceValue = distanceMatrixArrayClient.map((value) => value.distance.value);
```
Comment (vnersisyan-vineti) [inline src/callbacks/findReservationPrice.js:3]: separate these lines

File: src/callbacks/findReservationPrice.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  distanceValue.sort((a, b) => a - b);
+  let tripPrice = 0;
```
Comment (vnersisyan-vineti) [inline src/callbacks/findReservationPrice.js:5]: separate these lines

File: src/callbacks/findReservationPrice.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      break;
+    case 'First class':
+      tripPrice = (distanceValue[0] / 1000) * 200;
+      break;
+    case 'Cargo van':
+      tripPrice = (distanceValue[0] / 1000) * 250;
+      break;
+    default:
+      tripPrice = 0;
+  }
```
Comment (vnersisyan-vineti) [inline src/callbacks/findReservationPrice.js:21]: separate these lines

File: src/custom hooks/useDriversCoordinates.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from 'react';
+import axios from 'axios';
+import { baseUrl } from '../utils/configs';
+
+const useDriversCoordinates = () => {
+  const [driversPosition, setDriversPosition] = useState();
+  const [drivers, setDrivers] = useState();
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDriversCoordinates.js:7]: separate these lines

File: src/custom hooks/useDriversCoordinates.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  useEffect(() => {
+    const source = axios.CancelToken.source();
+    const config = { cancelToken: source.token };
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDriversCoordinates.js:10]: separate these lines

File: src/custom hooks/useDriversCoordinates.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    axios.get(`${baseUrl}/coordinates/drivers`, config)
+      .then((response) => {
+        setDrivers(response.data.drivers);
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDriversCoordinates.js:13]: separate these lines

File: src/custom hooks/useDriversCoordinates.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+        const coordinates = response.data.drivers.map((driver) => ({
+          lat: parseFloat(driver.latitude),
+          lng: parseFloat(driver.longitude),
+        }));
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDriversCoordinates.js:17]: separate these lines

File: src/custom hooks/useDriversCoordinates.js [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        setDriversPosition(coordinates);
+      })
+      .catch(() => {
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDriversCoordinates.js:20]: put console log of error here

File: src/custom hooks/useDriversCoordinates.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+      });
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDriversCoordinates.js:21]: separate these lines

File: src/custom hooks/useDriversCoordinates.js [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    return () => {
+      source.cancel();
+    };
+  }, []);
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDriversCoordinates.js:25]: separate these lines

File: src/custom hooks/useNearestDriver.js [hunk 15]
```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react';
+import axios from 'axios';
+import { baseUrl } from '../utils/configs';
+
+const useNearestDriver = (drivers, pickUpLocation, dropOffLocation,
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useNearestDriver.js:5]: place all variables in separate line please

File: src/custom hooks/useNearestDriver.js [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+  nearestDriverIndex, price, state) => {
+  const [message, setMessage] = useState();
+  let log;
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useNearestDriver.js:8]: separate these lines

File: src/custom hooks/useNearestDriver.js [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+  useEffect(() => {
+    const source = axios.CancelToken.source();
+    const config = { cancelToken: source.token };
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useNearestDriver.js:11]: separate these lines

File: src/custom hooks/useNearestDriver.js [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+      && nearestDriverIndex !== undefined && price) {
+      axios.post(`${baseUrl}/coordinates/trip_nearestdriver`, {
+        pickUpLocation,
+        dropOffLocation,
+        driverId: drivers[nearestDriverIndex].id,
+        passengerId: state.userId,
+        price,
+      }, config)
+        .then(() => {
+          clearInterval(log);
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useNearestDriver.js:22]: separate these lines

File: src/custom hooks/useNearestDriver.js [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+              axios.post(`${baseUrl}/coordinates/driverAssigned`, {
+                id: state.userId,
+              })
+                .then((response) => {
+                  if (response.data.message !== 'error') {
+                    setMessage('your driver is on the way');
+                    clearInterval(log);
+                  }
+                })
+                .catch(() => {
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useNearestDriver.js:34]: put console log of error here

File: src/custom hooks/useNearestDriver.js [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+                });
+            }, 3000,
+          );
+        })
+        .catch(() => {
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useNearestDriver.js:39]: put console log of error here

File: src/custom hooks/useNearestDriver.js [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+        });
+    }
+    return () => {
+      source.cancel();
+    };
+  }, [pickUpLocation, dropOffLocation, drivers, nearestDriverIndex, price]);
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useNearestDriver.js:45]: separate these lines

File: src/styles/map.css [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+
+#menuContainer{
+    width: 150px;
+    height: 166px;
+}
+
+#vehicleTypeContainer{
+    width: 300px;
+    height: 300px;
+}
```
Comment (vnersisyan-vineti) [inline src/styles/map.css:24]: empty line at the end of file

File: src/components/Client/Taxi.js [hunk 23]
```diff
@@ -15,139 +21,37 @@ const loader = new Loader({
...
-      {
-        origins: [pickUpLocation],
-        destinations: driversPosition,
-        travelMode: 'DRIVING',
-      }, callback1,
-    );
-  }
+  const findDriverIndex = (response) => {
+    setNearestDriverIndex(findNDI(response));
+  };
```
Comment (vnersisyan-vineti) [inline src/components/Client/Taxi.js:38]: separate these lines

File: src/components/Client/Taxi.js [hunk 24]
```diff
@@ -15,139 +21,37 @@ const loader = new Loader({
...
-      service.getDistanceMatrix(
-        {
-          origins: [pickUpLocation],
-          destinations: [dropOffLocation],
-          travelMode: 'DRIVING',
-        }, callback2,
-      );
-    }
-  }, [pickUpLocation, dropOffLocation]);
+  };
```
Comment (vnersisyan-vineti) [inline src/components/Client/Taxi.js:49]: separate these lines

File: src/custom hooks/useDistanceMatrix.js [hunk 25]
```diff
@@ -0,0 +1,26 @@
+import { useEffect } from 'react';
+
+const useDistanceMatrix = (origins, destinations, callback, google) => {
+  useEffect(() => {
+    if (origins && destinations) {
+      let dest = destinations;
+      let org = origins;
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDistanceMatrix.js:7]: separate these lines

File: src/custom hooks/useDistanceMatrix.js [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+      if (!Array.isArray(dest)) {
+        dest = [dest];
+      }
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDistanceMatrix.js:10]: separate these lines

File: src/custom hooks/useDistanceMatrix.js [hunk 27]
```diff
@@ -0,0 +1,26 @@
...
+      if (!Array.isArray(org)) {
+        org = [org];
+      }
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDistanceMatrix.js:13]: separate these lines

File: src/custom hooks/useDistanceMatrix.js [hunk 28]
```diff
@@ -0,0 +1,26 @@
...
+      const service = new google.maps.DistanceMatrixService();
```
Comment (vnersisyan-vineti) [inline src/custom hooks/useDistanceMatrix.js:14]: separate these lines

**Review by vnersisyan-vineti — CHANGES_REQUESTED** [review]
Overall good, but please run `npm run lint` before pushing. There will be a lot of lint errors, resolve them before pushing

File: src/callbacks/findNDI.js (hunk 1 above)
Comment (sevak-yer) [reply src/callbacks/findNDI.js:1]: done

**Comment by sevak-yer** [comment]
all done

**Pull request closed without merging**